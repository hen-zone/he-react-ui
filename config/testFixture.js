import createTestContext from 'react-cosmos-test/enzyme';

export default function testFixture(fixture, variant) {
  const { mount, getWrapper } = createTestContext({ fixture });
  const { component } = fixture;
  const { displayName, name } = component;
  mount();
  test(`<${displayName ||
    name} /> rendered correctly with ${variant} fixture`, () =>
    expect(getWrapper()).toMatchSnapshot());
}
