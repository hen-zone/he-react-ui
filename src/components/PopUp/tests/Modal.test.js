import fixture from '../fixtures/Modal.fixture.js'
import testFixture from '../../../../config/testFixture'

testFixture(fixture, 'Modal')
