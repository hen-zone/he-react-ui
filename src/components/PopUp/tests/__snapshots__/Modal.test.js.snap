// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopUp /> rendered correctly with Modal fixture 1`] = `
<PopUp
  modal={true}
  noPadding={false}
  onClose={[Function]}
  onOpen={[Function]}
  showing={true}
>
  <div
    className="outer showing"
  >
    <div
      className="overlay"
      onClick={[Function]}
    />
    <div
      className="popup popupWithPadding"
    >
      <div>
        <Paragraph>
          <p
            className="paragraph"
          >
            This modal popup cannot be closed without a parent interfering.
          </p>
        </Paragraph>
        <Paragraph>
          <p
            className="paragraph"
          >
            If only the showing property were set to false.
          </p>
        </Paragraph>
      </div>
    </div>
  </div>
</PopUp>
`;
